chia/wallet/nft_wallet/nft_off_chain.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from pathlib import Path
+from typing import Optional
+
+import requests
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.hash import std_hash
+from chia.wallet.nft_wallet.nft_info import NFTCoinInfo
+from chia.wallet.nft_wallet.uncurry_nft import UncurriedNFT
+
+PREFIX_HASH_LENGTH = 3
+CACHE_PATH_KEY = "nft_metadata_cache_path"
+DEFAULT_PATH = DEFAULT_ROOT_PATH / "nft_cache"
+DEFAULT_PATH.mkdir(parents=True, exist_ok=True)
```

Comment:
@altendky is this pattern ok? Creating a directory on module import?

Reply:
This probably isn't necessary since the directory is created in `write_off_chain_metadata()`, right?

Reply:
I don't like it, no.  Importing should not have side effects.

Separately from what we do here probably, there are proper locations for caches vs. tmp files vs. program data vs. config...  https://pypi.org/project/appdirs/ would be relevant if at some point we decided to follow the standards for that.